do/loadbalancers.go

```diff
@@ -179,6 +184,126 @@ func (l *loadbalancers) EnsureLoadBalancer(clusterName string, service *v1.Servi
 
 }
 
+func checkIfPortInRange(port int, fwPortRange string) bool {
+
+	fwPortList := strings.Split(fwPortRange, "-")
+	if fwPortList[0] == "all" {
+		return true
+	}
+	if len(fwPortList) == 1 {
+		fwPort, _ := strconv.Atoi(fwPortList[0])
+		if port == fwPort {
+			return true
+		}
+	} else {
+		fwLowerVal, _ := strconv.Atoi(fwPortList[0])
+		fwUpperVal, _ := strconv.Atoi(fwPortList[1])
+		if port >= fwLowerVal && port <= fwUpperVal {
+			return true
+		}
+	}
+
+	return false
+
+}
+
+func checkIfPortRequiresDeletion(port int, fwPortRange string) bool {
+
+	fwPortList := strings.Split(fwPortRange, "-")
+	if fwPortList[0] == "all" {
+		return false
+	}
+	if len(fwPortList) == 1 {
+		fwPort, _ := strconv.Atoi(fwPortList[0])
+		if port == fwPort {
+			return true
+		}
+	}
+
+	return false
+
+}
+
+func CreateFirewallsRuleRequest(sourcelb string, targetPort int) *godo.FirewallRulesRequest {
+	rr := &godo.FirewallRulesRequest{
+		InboundRules: []godo.InboundRule{
+			{
+				Protocol:  "tcp",
+				PortRange: strconv.Itoa(targetPort),
+				Sources: &godo.Sources{
+					LoadBalancerUIDs: []string{sourcelb},
+				},
+			},
+		},
+	}
+	return rr
+}
+
+type firewallRequest struct {
+	firewallId string
+	rules      *godo.FirewallRulesRequest
+}
+
+// EnsureFirewall ensures that the node firewalls have the correct rules for
+// the load balancer.
+//
+// EnsureFirewall will not modify service or nodes.
+func (l *loadbalancers) EnsureFirewall(lb *godo.LoadBalancer) error {
```

Comment:
The function name isn't really describing what this function does. It's ensuring that the firewall rules allow a load balancer. Not that a firewall exists.